tests/ashley/machina_extensions/forum_moderation/test_forum_moderation.py

```diff
@@ -0,0 +1,184 @@
+from typing import List
+
+from django.test import TestCase
+from machina.apps.forum_permission.shortcuts import assign_perm
+
+from ashley import SESSION_LTI_CONTEXT_ID
+from ashley.defaults import (
+    DEFAULT_FORUM_BASE_PERMISSIONS,
+    DEFAULT_FORUM_ROLES_PERMISSIONS,
+)
+from ashley.factories import (
+    ForumFactory,
+    LTIContextFactory,
+    PostFactory,
+    TopicFactory,
+    UserFactory,
+)
+from ashley.machina_extensions.forum_moderation.forms import TopicMoveForm
+
+
+class ForumModerationTestViewForm(TestCase):
+    def _init_forum(self, forum, context):
+        """
+        Initialize a forum.
+
+        This method sets default permissions according to the LTI context that
+        initiated the creation of the forum.
+        """
+        forum.lti_contexts.add(context)
+        self._assign_permissions(
+            forum,
+            context.get_base_group(),
+            DEFAULT_FORUM_BASE_PERMISSIONS,
+        )
+        # pylint: disable=no-member
+        for role, perms in DEFAULT_FORUM_ROLES_PERMISSIONS.items():
+            self._assign_permissions(forum, context.get_role_group(role), perms)
+
+    @staticmethod
+    def _assign_permissions(forum, group, permissions: List[str]):
+        """Grant a list of permissions for a group on a specific forum."""
+        for perm in permissions:
+            assign_perm(perm, group, forum, True)
+
+    def test_forum_moderation_list_forums_view(self):
+        """
+        Creates forum in different LTIContext, make sure user can't move topics into it,
+        even if he has access to this forum.
+        """
+        user = UserFactory()
+
+        lti_context1 = LTIContextFactory(lti_consumer=user.lti_consumer)
+        lti_context2 = LTIContextFactory(lti_consumer=user.lti_consumer)
+        forum1Lti1 = ForumFactory(name="Forum1")
+        forum2Lti1 = ForumFactory(name="Forum2")
+        forum3Lti2 = ForumFactory(name="Forum3")
+        forum4Lti2 = ForumFactory(name="Forum4")
+
+        # User is instructor in both forums
+        lti_context1.sync_user_groups(user, ["instructor"])
+        lti_context2.sync_user_groups(user, ["instructor"])
+
+        # assign permission to the group for this forum
+        self._init_forum(forum1Lti1, lti_context1)
+        self._init_forum(forum2Lti1, lti_context1)
+        self._init_forum(forum3Lti2, lti_context2)
+        self._init_forum(forum4Lti2, lti_context2)
+
+        # Creates a post for a topic part of lti_context1
+        topicForumLti1 = TopicFactory(forum=forum1Lti1)
+        PostFactory(
+            topic=topicForumLti1,
+        )
+
+        # Creates a post for a topic part of lti_context2
```

Comment:
```suggestion
        # Create a post for a topic part of lti_context2
```

+ more occurences in the following :wink: